pkg/kv/postgres/store.go

```diff
@@ -172,27 +212,37 @@ func (s *Store) SetIf(ctx context.Context, key, value []byte, valuePredicate kv.
 	return nil
 }
 
-func (s *Store) Delete(ctx context.Context, key []byte) error {
+func (s *Store) Delete(ctx context.Context, partitionKey, key []byte) error {
+	if len(partitionKey) == 0 {
+		return kv.ErrMissingPartitionKey
+	}
 	if len(key) == 0 {
 		return kv.ErrMissingKey
 	}
-	_, err := s.Pool.Exec(ctx, `DELETE FROM `+s.Params.SanitizedTableName+` WHERE key=$1`, key)
+	_, err := s.Pool.Exec(ctx, `DELETE FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 AND key=$2`, partitionKey, key)
 	if err != nil {
 		return fmt.Errorf("%s: %w (key=%v)", err, kv.ErrOperationFailed, key)
 	}
 	return nil
 }
 
-func (s *Store) Scan(ctx context.Context, start []byte) (kv.EntriesIterator, error) {
+func (s *Store) Scan(ctx context.Context, partitionKey, start []byte) (kv.EntriesIterator, error) {
 	var (
 		rows pgx.Rows
 		err  error
 	)
-	if start == nil {
-		rows, err = s.Pool.Query(ctx, `SELECT key,value FROM `+s.Params.SanitizedTableName+` ORDER BY key`)
-	} else {
-		rows, err = s.Pool.Query(ctx, `SELECT key,value FROM `+s.Params.SanitizedTableName+` WHERE key >= $1 ORDER BY key`, start)
+
+	switch {
+	case start == nil && len(partitionKey) == 0:
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` ORDER BY key`)
+	case start != nil && len(partitionKey) == 0:
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE key >= $1 ORDER BY key`, start)
```

Comment:
Is scanning without specifying a partition key (scanning all partitions) a real use case?
The reason I'm asking is that DynamoDB does not support it, and in order to get it done we will have to explicitly iterate over all partitions.